include/oboe/FullDuplexStream.h

```diff
@@ -53,11 +53,21 @@ class FullDuplexStream : public AudioStreamDataCallback {
     virtual ~FullDuplexStream() = default;
 
     /**
-     * Sets the input stream. Calling this is mandatory.
+     * Sets the input stream.
      *
+     * @deprecated Call setInputStream(std::shared_ptr<AudioStream> &stream) instead.
      * @param stream the output stream
      */
     void setInputStream(AudioStream *stream) {
+        mInputStream = std::shared_ptr<AudioStream>(stream);
```

Comment:
This is dangerous because it could lead to a double free.
Robert suggested keeping two pointers internally, one raw, one shared. 
Use the shared_ptr first.

Have two APIs:
get/setInputStream() for raw pointers. DEPRECATED
get/setSharedInputStream()  for shared_ptr. RECOMMENDED

Reply:
Fixed in newest commit. Thanks for the great suggestion!